python/ray/data/datasource/datasource.py

```diff
@@ -158,19 +184,42 @@ def make_block(start: int, count: int) -> Block:
                 schema = int
             else:
                 raise ValueError("Unsupported block type", block_format)
+            meta = BlockMetadata(
+                num_rows=count,
+                size_bytes=8 * count,
+                schema=schema,
+                input_files=None)
             read_tasks.append(
                 ReadTask(
-                    lambda i=i, count=count: make_block(i, count),
-                    BlockMetadata(
-                        num_rows=count,
-                        size_bytes=8 * count,
-                        schema=schema,
-                        input_files=None)))
+                    lambda i=i, count=count, meta=meta: [(
+                        ray.put(make_block(i, count), _owner=owner), meta)],
```

Comment:
Ok cool, just wanted to call that out as a risk for the fault-tolerance target. The detached actor/GCS owner case seems easier to support than arbitrary owners, since you don't have to deal with owner death for the former, so the risk doesn't seem too high to me.